scripts/language_model/word_language_model.py

```diff
@@ -222,96 +235,38 @@ def evaluate(data_source, batch_size, ctx=None):
         data = data.as_in_context(ctx)
         target = target.as_in_context(ctx)
         output, hidden = model(data, hidden)
-        hidden = detach(hidden)
+        hidden = nlp.model.detach(hidden)
         L = loss(output.reshape(-3, -1),
                  target.reshape(-1,))
         total_L += mx.nd.sum(L).asscalar()
         ntotal += L.size
     return total_L / ntotal
 
-
-def forward(inputs, begin_state=None):
-    """Implement forward computation using awd language model.
+def criterion(output, target, encoder_hs, dropped_encoder_hs):
```

Comment:
we don't use the name `criterion` in mxnet. use `loss` instead.

Reply:
well, loss is the cross entropy in this script. Criterion means a user defined loss. I think it is fine to use criterion in the script.